src/GridFieldEditableColumns.php

```diff
@@ -104,6 +100,10 @@ public function getColumnContent($grid, $record, $col)
         $field->setName($this->getFieldName($field->getName(), $grid, $record));
         $field->setValue($value);
 
+        if (!$record->canEdit() || $grid->isReadonly()) {
```

Comment:
tiny optimisation may bee to check `isReadOnly` first as I assume that's a much less-intensive operation in the case that users can edit (typical) and that it is read only?

Reply:
Fair enough - force-pushing a fix.